api/test-context/src/test_context.rs

```diff
@@ -382,6 +405,57 @@ impl TestContext {
         ret
     }
 
+    pub async fn read_resource(
+        &self,
+        account_address: &AccountAddress,
+        resource: &str,
+    ) -> Option<Value> {
+        let request = format!("/accounts/{}/resources", account_address);
```

Comment:
Why not use the API for fetching a specific resource instead of fetching them all then filtering for the right one?

Reply:
it throws an error if the resource doesn't exist.